File: CS/WindowsApplication1/Custom Ribbon Control/MyRibbonPanelViewInfo.cs
using DevExpress.XtraBars.Ribbon;
using DevExpress.XtraBars.Ribbon.ViewInfo;
using WindowsApplication1.Custom_Ribbon_Control;

namespace WindowsApplication1 {
    public class MyRibbonPanelViewInfo : RibbonPanelViewInfo {
        public MyRibbonPanelViewInfo(RibbonViewInfo viewInfo)
            : base(viewInfo) {
        }
        protected override RibbonPageGroupViewInfo CreateGroupViewInfo(RibbonPageGroup group) {
            return new MyRibbonPageGroupViewInfo(ViewInfo, group);
        }
        protected override RibbonPanelLayoutCalculator CreatePanelLayoutCalculator() {
            return new MyRibbonPanelComplexLayoutCalculator(this);
        }
    }
}

File: VB/WindowsApplication1/Custom Ribbon Control/MyRibbonPanelViewInfo.vb
Imports DevExpress.XtraBars.Ribbon
Imports DevExpress.XtraBars.Ribbon.ViewInfo
Imports WindowsApplication1.Custom_Ribbon_Control

Namespace WindowsApplication1
    Public Class MyRibbonPanelViewInfo
        Inherits RibbonPanelViewInfo

        Public Sub New(ByVal viewInfo As RibbonViewInfo)
            MyBase.New(viewInfo)
        End Sub
        Protected Overrides Function CreateGroupViewInfo(ByVal group As RibbonPageGroup) As RibbonPageGroupViewInfo
            Return New MyRibbonPageGroupViewInfo(ViewInfo, group)
        End Function
        Protected Overrides Function CreatePanelLayoutCalculator() As RibbonPanelLayoutCalculator
            Return New MyRibbonPanelComplexLayoutCalculator(Me)
        End Function
    End Class
End Namespace
